genisys/modules/os_file_download.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing_extensions import Self
+import requests
+import tarfile
+import os
+from genisys.modules import base
+
+class OSDownload(base.Module):
+    """
+    This module handles the downloading of OS files necessary for PXE booting a client machine.
+    - vmlinuz (Kernel) & initrd/initramfs (Initial Ram Disk)
+    - pxelinux.cfg
+    - pxelinux.0 or GRUB (Depends on OS/Version)
+    Link to Debian files (amd64): https://deb.debian.org/debian/dists/bookworm/main/installer-amd64/current/images/
+    """
+
+    def __init__(self: Self, config) -> None:
+        self.config = config
+        self.config["network"] = config["Network"]
+
+        self.DEBIAN_TAR_FILENAME = "netboot.tar.gz"
+        self.DEBIAN_TAR_LINK = f"https://deb.debian.org/debian/dists/{self.config['OS']['version-name']}/main/installer-{self.config['OS']['target-architecture']}/current/images/netboot/netboot.tar.gz"
+    # end __init__
+
+    def install(self: Self, chroot: Path = ...):
+        tftp_directory = self.config["network"]["tftp_directory"]
+
+        try:
+            response = requests.get(self.DEBIAN_TAR_LINK, allow_redirects=True, timeout=15)
+            response.raise_for_status()  # Check for request errors
+
+            tmp_directory = Path("/tmp")
```

Comment:
Use https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir for better cross-platform support

Reply:
Addressed in commit [7eadcab](https://github.com/gingrm02/genisys/pull/16/commits/7eadcab61ad4248fc9f7b5294e74ad2c5120be73)